src/turn.c

```diff
@@ -112,13 +112,13 @@ static int find_ordered_transaction_id_rec(turn_entry_t *const ordered_transacti
                                            const uint8_t *transaction_id, int begin, int end) {
 	int d = end - begin;
 	if (d <= 0)
-		return begin;
+		return -1;
 
-	int pivot = d / 2;
+	int pivot = begin + d / 2;
 	const turn_entry_t *entry = ordered_transaction_ids[pivot];
 	int ret = memcmp(transaction_id, entry->transaction_id, STUN_TRANSACTION_ID_SIZE);
 	if (ret < 0)
-		return find_ordered_transaction_id_rec(ordered_transaction_ids, transaction_id, 0, pivot);
+		return find_ordered_transaction_id_rec(ordered_transaction_ids, transaction_id, begin, pivot);
```

Comment:
Same, could you please also fix the binary search for the channel?

Reply:
done